packages/bp/src/admin/workspace/collaborators/collaborators-router.ts

```diff
@@ -41,15 +41,36 @@ class CollaboratorsRouter extends CustomAdminRouter {
       this.needPermissions('read', this.resource),
       this.asyncMiddleware(async (req, res) => {
         const filterRoles = req.query.roles && req.query.roles.split(',')
+        const filterAuthStrategies: string[] | undefined =
+          req.query.authStrategies && req.query.authStrategies.split(',')
         const allUsers = await this.authService.getAllUsers()
-        const workspaceUsers = await this.workspaceService.getWorkspaceUsers(req.workspace!)
-        const available = _.filter(allUsers, x => !_.find(workspaceUsers, x)) as WorkspaceUser[]
+        if (!req.workspace) {
+          throw new BadRequestError('Workspace must be defined to list available users')
+        }
 
-        return sendSuccess(
-          res,
-          'Retrieved available users',
-          filterRoles ? available.filter(x => filterRoles.includes(x.role)) : available
-        )
+        const workspaceUsers = await this.workspaceService.getWorkspaceUsers(req.workspace)
+
+        let allWorkspaceUsers: WorkspaceUser[] = []
+        for (const w of await this.workspaceService.getWorkspaces()) {
+          const users = (await this.workspaceService.getWorkspaceUsers(w.id)) as WorkspaceUser[]
+          allWorkspaceUsers = allWorkspaceUsers.concat(users)
+        }
```

Comment:
I'm not sure of this implementation why do you fetch all users from all workspaces ? 

If you look at the previous implementation what we were returning was a filtered list of `allUsers`

Reply:
the allUsers object doesn't list user roles, I needed to pull them from workspaces to get their associated roles in those workspaces so I could match against that. Is there a better way to do this?